src/runtime/crt/common/crt_runtime_api.c

```diff
@@ -38,7 +38,7 @@
 
 static char g_last_error[1024];
 
-void TVMAPISetLastError(const char* msg) { strncpy(g_last_error, msg, sizeof(g_last_error)); }
+void TVMAPISetLastError(const char* msg) { strncpy(g_last_error, msg, (sizeof(g_last_error)-1)); }
```

Comment:
This won't null-terminate the string if msg >= g_last_error in size.  You can just add g_last_error[sizeof(g_last_error)-1] = 0; to make sure the target string is always null-terminated.